src/schedule/dma_single_chan_domain.c

```diff
@@ -0,0 +1,491 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Tomasz Lauda <tomasz.lauda@linux.intel.com>
+
+#include <sof/audio/component.h>
+#include <sof/bit.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/drivers/timer.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/notifier.h>
+#include <sof/platform.h>
+#include <sof/schedule/ll_schedule_domain.h>
+#include <sof/schedule/schedule.h>
+#include <sof/schedule/task.h>
+#include <ipc/topology.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define DMA_DOMAIN_OWNER_INVALID	0xFFFFFFFF
+
+struct dma_domain_data {
+	struct notifier notifier;
+	int irq;
+	struct dma_chan_data *channel;
+	void (*handler)(void *arg);
+	void *arg;
+};
+
+struct dma_domain {
+	struct dma *dma_array;	/* pointer to scheduling DMAs */
+	uint32_t num_dma;	/* number of scheduling DMAs */
+	uint32_t owner;		/* core owning the scheduling channel */
+	bool channel_changed;	/* true if we needed to re-register */
+
+	/* data per core */
+	struct dma_domain_data data[PLATFORM_CORE_COUNT];
+};
+
+struct ll_schedule_domain_ops dma_single_chan_domain_ops;
+
+static void dma_single_chan_domain_enable(struct ll_schedule_domain *domain,
+					  int core);
```

Comment:
Is it misaligned or GH shows it incorrectly?

Reply:
Looks like tabs are at it again. I think they're aligned.